packages/playwright-core/src/server/injected/reactSelectorEngine.ts

```diff
@@ -144,9 +144,16 @@ function findReactRoots(root: Document | ShadowRoot, roots: ReactVNode[] = []):
   const walker = document.createTreeWalker(root, NodeFilter.SHOW_ELEMENT);
   do {
     const node = walker.currentNode;
+    // ReactDOM Legacy client API:
     // @see https://github.com/baruchvlz/resq/blob/5c15a5e04d3f7174087248f5a158c3d6dcc1ec72/src/utils.js#L329
-    if (node.hasOwnProperty('_reactRootContainer'))
+    if (node.hasOwnProperty('_reactRootContainer')) {
```

Comment:
Turns out all that stuff was already legacy in React 17: https://github.com/facebook/react/blob/4985bb0a80f5cbeaa61d21a7daf7da5ecff2d892/packages/react-dom/src/client/ReactDOMLegacy.js#L193-L196